lib/repositories/environment.py

```diff
@@ -1,101 +1,138 @@
+import logging
+from datetime import datetime
 from typing import Union
-from pymongo.results import InsertOneResult
-from pymongo.results import DeleteResult
+from lib.repositories import parse_error
 from lib.models.environment import Env
 from lib.repositories.repo import Repository
 
 
 class EnvRepository(Repository):
     """
-    Environment repository
+    Enables database CRUD operations with models.Env
 
     Init Attributes:
-        environment: Env object
-        env_id: Environment id
+        environment: models.Env
+        env_id: str
 
-    Enables CRUD operations on environment objects
     """
 
     def __init__(self, environment: Env = None, env_id: str = None):
         super().__init__("environments")
-        self.environment = environment
+        self._env = environment
         if env_id:
-            self.env_id = env_id
+            self._env_id = env_id
         else:
-            self.env_id = self.environment.__hash__()
+            self._env_id = hash(self._env)
```

Comment:
Agreed!

For now, since we are in Beta the odds of hash colision are very low. But we do have to implement a custom hashing mechanism in the future, one that is not session-specific.

Thanks for bringing this up!